pkg/timeseries/dataset/point.go

```diff
@@ -45,6 +47,61 @@ func (p *Point) Clone() Point {
 	return clone
 }
 
+// Equal returns true if p and p2 are exactly equal
+func PointsAreEqual(p1, p2 Point) bool {
+	if p1.Epoch != p2.Epoch || p1.Size != p2.Size ||
+		len(p1.Values) != len(p2.Values) {
+		return false
+	}
+	for i, v := range p1.Values {
+		switch t := v.(type) {
```

Comment:
have you considered this generic func from the stdlib: https://pkg.go.dev/cmp#Compare